lib/fluent/plugin/buffer.rb

```diff
@@ -444,6 +451,12 @@ def write(metadata_and_data, format: nil, size: nil, enqueue: false)
             end
             chunk.mon_exit rescue nil # this may raise ThreadError for chunks already committed
           end
+          if @delay_buffer_chunk_overflow_error
+            # Notify delayed chunk overflow here
+            message = synchronize { @delay_buffer_chunk_overflow_error }
```

Comment:
Should wrap the whole of the transaction of `@delay_buffer_chunk_overflow_error` instead of each lines.